### ISSUE TITLE
eval_image_classifier.py Number of images evaluated wrong

### ISSUE THREAD

**Issue by nikhileshsharma91** [tensorflow/models#2122]
I followed the tutorial to finetune the inception model for the Flowers dataset.

The flowers dataset has 350 validation images specified in the flowers.py file.

But when I ran eval_image_classifier.py and modified it to print the number of TP,FP,TN,FN

The results:
```
 tensorflow/core/kernels/logging_ops.cc:79] eval/TrueNegatives[64]
I tensorflow/core/kernels/logging_ops.cc:79] eval/TruePositives[286]
I tensorflow/core/kernels/logging_ops.cc:79] eval/FalsePositives[2]
I tensorflow/core/kernels/logging_ops.cc:79] eval/FalseNegatives[48]
```
If you add the them up, its a total of 400. But the number of validation images is 350.

I did fine tuning for my custom dataset, where the validation images were 150 with only two classes.

The results were:
```
I tensorflow/core/kernels/logging_ops.cc:79] eval/TruePositives[11]
I tensorflow/core/kernels/logging_ops.cc:79] eval/TrueNegatives[155]
I tensorflow/core/kernels/logging_ops.cc:79] eval/FalsePositives[4]
I tensorflow/core/kernels/logging_ops.cc:79] eval/FalseNegatives[30]
I tensorflow/core/kernels/logging_ops.cc:79] eval/Accuracy[0.83]
I tensorflow/core/kernels/logging_ops.cc:79] eval/AreaUnderCurve[0.62156773]
```

If you add them up its comes out to be 200.

Why is this happening? Where are the additional 50 images coming from?

Is there a way to modify eval_image_classifier.py to print the name of the validation images with its predictions and labels?

**Comment by aureosun** [comment]
Hello,
I am confused at the use of tf-slim for image classification,how can I make the Creating a TF-Slim Dataset Descriptor? Could you please tell me the details?

**Comment by nikhileshsharma91** [comment]
You would like to know how to create a data descriptor for a custom dataset?

**Comment by nikhileshsharma91** [comment]
I asked this question on stackoverflow
[https://stackoverflow.com/questions/45538979/tensorflow-models-slim-eval-image-classifier-py-number-of-images-evaluated-wrong/45540193?noredirect=1#comment78044213_45540193]

I think it should throw an error if the batch_size is set wrong. For example the flower dataset has 350 images and the default batch_size is 100. 

` num_batches = math.ceil(dataset.num_samples / float(FLAGS.batch_size))`

This line of code rounds it off. 350/100 is equals 4. So the num_batches is 4 with the default batch_size=100 that makes it 400 images in total. But the validation set only has 350 images. It should definitely throw an error.

**Comment by aureosun** [comment]
Sir,Would you please send me your code ? that print the number of TP,FP,TN,FN?I met the same problem and I use my own data,and adjust the batchsize to a num that can be divided with no remainder.my email is chearain@sina.com

**Comment by lichuanzong** [comment]
Hello!
Can you tell me how you got the number of TP,FP,TN,FN?
If you can, can you give me the code？Thank you,very much!

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
